segment.go

```diff
@@ -17,7 +17,8 @@ type SegmentID uint32
 type Segment struct {
 	// msg associated with this segment. A Message instance m maintains the
 	// invariant that that all m.segs[].msg == m.
-	msg  *Message
+	msg *Message
+
 	id   SegmentID
```

Comment:
Something to note:  external packages cannot currently provide their own `Arena` implementations because the `data` and `id` fields are not exported.

How do you feel about bundling a fix with this PR?  The simplest approach I can think of is to turn `Segment` into an interface.

Reply:
>How do you feel about bundling a fix with this PR? The simplest approach I can think of is to turn Segment into an interface.

Happy to do this in a separate PR too, seeing as you're trying to minimize breaking changes to `Arena`.

Reply:
Yeah, my intention is to do it over a separate PR. Changing `Segment` into an interface is one possibility (which would eventually allow stuff like a memory-mapped arena impl), but I want to do it over a separate PR in order to evaluate the performance impact.

There are also some additional types that need to be exported as well (such as `address`).